jetty-server/src/main/java/org/eclipse/jetty/server/handler/FileBufferedResponseHandler.java

```diff
@@ -225,7 +216,21 @@ protected void onCompleteFailure(Throwable cause)
                     dispose();
                     callback.failed(cause);
                 }
+
+                @Override
+                public String toString()
+                {
+                    final StringBuilder sb = new StringBuilder("FileBufferedInterceptor[");
+                    sb.append("fileLength=").append(fileLength);
+                    sb.append(", pos=").append(_pos);
+                    sb.append(", last=").append(_last);
+                    sb.append(']');
+                    return sb.toString();
+                }
             };
+            // once icb is GC'd, cleanup the File.
+            if (_filePath != null)
+                PhantomCleaner.register(icb, new PhantomCleaner.FileDispose(_filePath));
```

Comment:
Do we only want to do delayed deletion on windows? or do we want to use the same mechanism everywhere?

Perhaps we could try to delete here and if that fails then register the cleaner?

Reply:
Also is the icb really the right object?  Surely it should be the buffer?   icb doesn't even have a reference to the buffer.

Reply:
@gregw it cannot be the buffer because we may iterate and have multiple buffers. Using the ICB guarantees that when the ICB is done, then all the buffers are also done and we register with the cleaner only once.
It is also critical that we register a reference to an object that is not strongly referenced, and the ICB reference satisfies this condition.

This mechanism will delay the file deletion in all cases, not just Windows.
Trying to delete and if fails use the cleaner will introduce a race as the cleaner may run at any moment... but `Files.deleteIfExists()` should take care of that.